src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -262,26 +280,121 @@ private void signUpButtonClicked(ActionEvent e) {
         }
     }
 
+    /*
+        Main Chat
+    */
     private void addConversation(ActionEvent e) {
-        // popup for the user to add a conversation
-        TextInputDialog dialog = new TextInputDialog("Name your conversation!");
-        dialog.setTitle(" "); // make the title blank
-        dialog.setHeaderText("Create your conversation");
+        if (clientContext.user.hasCurrent()) {
+            // popup for the user to add a conversation
+            TextInputDialog dialog = new TextInputDialog("Name your conversation!");
+            dialog.setTitle(" "); // make the title blank
+            dialog.setHeaderText("Create your conversation");
+
+            // get input and add the name of the convo
+            String name = dialog.showAndWait().get();
+
+            // TODO: check for duplicate conversations & handle them
+
+            if (name != null && name.length() > 0) {
+                clientContext.conversation.startConversation(name, clientContext.user.getCurrent().id);
+                convoList.addAll(name);
+            }
+        } else {
+            // user is not signed in
+            Alert alert = new Alert(AlertType.ERROR);
+            alert.setHeaderText("You're not signed in!");
+            alert.showAndWait();
+        }
+    }
 
-        // get input and add the name of the convo
-        String name = dialog.showAndWait().get();
+    private void selectConversation(MouseEvent e) {
+        // set the conversation title
+        int index = conversations.getSelectionModel().getSelectedIndex();
+        String data = conversations.getSelectionModel().getSelectedItem();
+        ConversationSummary cs = ChatGuiFX.this.lookupByTitle(data, index);
+
+        clientContext.conversation.setCurrent(cs);
+        update(cs);
+    }
 
-        // TODO: check for duplicate conversations & handle them
+    // Locate the Conversation object for a selected title string.
+    // index handles possible duplicate titles.
+    private ConversationSummary lookupByTitle(String title, int index) {
 
-        if (name != null && name.length() > 0) {
-            // TODO: users need to be added to the server first? So I can add the conversation to that user???
-            //clientContext.conversation.startConversation(name, clientContext.user.getCurrent().id);
-            convoList.addAll(name);
+      int localIndex = 0;
+      for (final ConversationSummary cs : clientContext.conversation.getConversationSummaries()) {
+        if ((localIndex >= index) && cs.title.equals(title)) {
+          return cs;
         }
+        localIndex++;
+      }
+      return null;
     }
 
-    private void selectConversation(MouseEvent e) {
-        // set the conversation title
-        chatTitle.setText(conversations.getSelectionModel().getSelectedItem());
+    // External agent calls this to trigger an update of this panel's contents.
+    public void update(ConversationSummary owningConversation) {
```

Comment:
This is a pretty generic function name. What is being updated? What triggers this function? It's public, so is this being called from an external class? What is the parameter? Consider using Javadoc style comments, especially for public functions and class-level variables.